restapi/src/main/java/io/stargate/web/resources/RowResource.java

```diff
@@ -400,12 +400,12 @@ private static Object filterToValue(Object val, String column, Table tableData)
   private List<Where<?>> buildWhereClause(PathSegment id, Table tableMetadata) {
     List<String> values = idFromPath(id);
 
-    final List<Column> keys = tableMetadata.partitionKeyColumns();
-    if (keys.size() != values.size()) {
-      throw new IllegalArgumentException("not enough partition keys provided");
+    final List<Column> keys = tableMetadata.primaryKeyColumns();
+    if (keys.size() < values.size()) {
```

Comment:
Note: I see tests covering using pk and pk+ck, so possibly this comment is incorrect. Feel free to ignore.

Is this correct wrt to acceptance criteria?
If we have `(pk0, pk1, pk2, ck0, ck1)` then:
pk0, pk1, pk2 => whole partition
pk0, pk1, pk2, ck0 => partition filtered by ck0
pk0, pk1, pk2, ck0, ck1 => partition filtered by ck0, ck1

Similar to CQL:
```cqlsh> select * from test.c29 where pk0=1 and pk1='one' and pk2=-1;

 pk0 | pk1 | pk2 | ck0 | ck1   | st           | v
-----+-----+-----+-----+-------+--------------+----
   1 | one |  -1 |  10 | jeden | 999999999999 |  9
   1 | one |  -1 |  20 | jeden | 999999999999 | 19

(2 rows)
cqlsh> select * from test.c29 where pk0=1 and pk1='one' and pk2=-1 and ck0=10;

 pk0 | pk1 | pk2 | ck0 | ck1   | st           | v
-----+-----+-----+-----+-------+--------------+---
   1 | one |  -1 |  10 | jeden | 999999999999 | 9
```

Reply:
Correct. Shooting for same behavior as cql